File: .gt-tmp/score.py
#!/usr/bin/env python3
import json, math, os, glob, sys
from datetime import datetime, timezone

TMPDIR = "/home/runner/work/aeon/aeon/.gt-tmp"
NOW = datetime(2026, 5, 29, 22, 14, 0, tzinfo=timezone.utc)

# Load all pool files
FILES = {
    "gt-global.json": "global",
    "gt-new.json": "new",
    "gt-solana-trend.json": "sol-trend",
    "gt-solana-vol.json": "sol-vol",
    "gt-eth-trend.json": "eth-trend",
    "gt-eth-vol.json": "eth-vol",
    "gt-base-trend.json": "base-trend",
    "gt-base-vol.json": "base-vol",
    "gt-bsc-trend.json": "bsc-trend",
    "gt-bsc-vol.json": "bsc-vol",
    "gt-arbitrum-trend.json": "arb-trend",
    "gt-arbitrum-vol.json": "arb-vol",
}

all_pools = []
src_status = {}
for fname, label in FILES.items():
    path = os.path.join(TMPDIR, fname)
    if not os.path.exists(path):
        src_status[label] = "missing"
        continue
    try:
        with open(path) as f:
            data = json.load(f)
        if "data" not in data:
            src_status[label] = "fail"
            continue
        for p in data["data"]:
            p["_source"] = label
            all_pools.append(p)
        src_status[label] = "ok"
    except Exception as e:
        src_status[label] = f"err: {e}"

print(f"Loaded {len(all_pools)} raw pool entries from {len([s for s in src_status.values() if s=='ok'])}/{len(FILES)} sources")

def fnum(d, *keys, default=0.0):
    cur = d
    for k in keys:
        if cur is None or not isinstance(cur, dict) or k not in cur:
            return default
        cur = cur[k]
    if cur is None: return default
    try: return float(cur)
    except: return default

def inum(d, *keys, default=0):
    v = fnum(d, *keys, default=None)
    if v is None: return default
    try: return int(v)
    except: return default

# Dedup by base_token id, keep highest volume_usd.h24
by_token = {}
for p in all_pools:
    attrs = p.get("attributes", {})
    rels = p.get("relationships", {})
    bt = rels.get("base_token", {}).get("data", {}).get("id")
    if not bt:
        continue
    vol = fnum(attrs, "volume_usd", "h24")
    if bt not in by_token or vol > fnum(by_token[bt].get("attributes", {}), "volume_usd", "h24"):
        by_token[bt] = p

print(f"After dedup: {len(by_token)} unique tokens")
pre_gate = len(by_token)

# Gate
gate_rejections = {"thin-vol":0, "negative":0, "thin-liq":0, "dumping":0, "honeypot":0, "too-new":0, "rug-like":0}
survivors = []
for bt, p in by_token.items():
    a = p.get("attributes", {})
    vol_h24 = fnum(a, "volume_usd", "h24")
    pct_h24 = fnum(a, "price_change_percentage", "h24")
    liq = fnum(a, "reserve_in_usd")
    buys = inum(a, "transactions", "h24", "buys")
    sells = inum(a, "transactions", "h24", "sells")
    pca = a.get("pool_created_at")
    age_h = 9999.0
    if pca:
        try:
            dt = datetime.fromisoformat(pca.replace("Z", "+00:00"))
            age_h = (NOW - dt).total_seconds() / 3600.0
        except: pass

    if vol_h24 < 50000: gate_rejections["thin-vol"] += 1; continue
    if pct_h24 <= 0: gate_rejections["negative"] += 1; continue
    if liq < 10000: gate_rejections["thin-liq"] += 1; continue
    if buys > 0 and sells / max(buys, 1) > 10: gate_rejections["dumping"] += 1; continue
    if sells > 0 and buys / max(sells, 1) > 50: gate_rejections["honeypot"] += 1; continue
    if age_h < 1.0 and vol_h24 < 100000: gate_rejections["too-new"] += 1; continue
    if pct_h24 > 10000: gate_rejections["rug-like"] += 1; continue

    p["_meta"] = dict(vol_h24=vol_h24, pct_h24=pct_h24, liq=liq, buys=buys, sells=sells, age_h=age_h)
    survivors.append(p)

print(f"Post-gate: {len(survivors)} pools")
print(f"Gate rejections: {gate_rejections}")

# Score
def clamp(x, lo, hi): return max(lo, min(hi, x))

for p in survivors:
    a = p.get("attributes", {})
    m = p["_meta"]
    pct = m["pct_h24"]
    vol = m["vol_h24"]
    liq = m["liq"]
    pct_h1 = fnum(a, "price_change_percentage", "h1")
    buys = m["buys"]; sells = m["sells"]

    pct_pts = clamp(pct / 500.0, 0, 1)
    vol_pts = clamp(math.log10(vol + 1) / 7.0, 0, 1)
    liq_pts = clamp(math.log10(liq + 1) / 6.0, 0, 1)
    mom_pts = clamp((pct_h1 + 50.0) / 100.0, 0, 1)
    skew_pts = clamp(buys / max(buys + sells, 1), 0, 1)

    score = 40*pct_pts + 25*vol_pts + 15*liq_pts + 10*mom_pts + 10*skew_pts
    p["_meta"]["score"] = score
    p["_meta"]["pct_h1"] = pct_h1
    p["_meta"]["pct_h6"] = fnum(a, "price_change_percentage", "h6")

# Tag
def tag(p):
    m = p["_meta"]
    if m["liq"] >= 1_000_000 and m["vol_h24"] >= 1_000_000:
        return "DEEP-LIQ"
    if m["age_h"] <= 48 and m["vol_h24"] >= 250_000:
        return "BREAKOUT"
    if m["pct_h1"] > 2 and m["pct_h24"] > 50:
        return "CONTINUATION"
    if m["pct_h1"] < -5 and m["pct_h24"] > 0:
        return "REVERSAL"
    return "MICRO-SPEC"

for p in survivors:
    p["_meta"]["tag"] = tag(p)

# Sort by score
survivors.sort(key=lambda p: -p["_meta"]["score"])

# Verdict from top 5
top5 = survivors[:5]
deep_count = sum(1 for p in top5 if p["_meta"]["tag"] == "DEEP-LIQ")
cont_count = sum(1 for p in top5 if p["_meta"]["tag"] == "CONTINUATION")
micro_count = sum(1 for p in top5 if p["_meta"]["tag"] in ("MICRO-SPEC", "BREAKOUT"))

if len(survivors) < 5:
    verdict = "SLEEPY"
elif deep_count >= 2:
    verdict = "STRONG"
elif deep_count == 1 or cont_count >= 2:
    verdict = "MIXED"
else:
    verdict = "SPECULATIVE"

print(f"\n=== VERDICT: {verdict} ===")
print(f"Top 5 tags: deep_liq={deep_count}, continuation={cont_count}, micro+breakout={micro_count}")

print("\n=== TOP 15 ===")
for i, p in enumerate(survivors[:15], 1):
    a = p.get("attributes", {})
    m = p["_meta"]
    name = a.get("name", "?")
    network = p.get("relationships", {}).get("network", {}).get("data", {}).get("id") or (p.get("id","").split("_")[0] if "_" in p.get("id","") else "?")
    pool_id = p.get("id", "?")
    print(f"{i:2d}. [{m['tag']:12s}] {name:35s} ({network:9s}) +{m['pct_h24']:.0f}% — score {m['score']:.1f}, vol ${m['vol_h24']/1e6:.1f}m, liq ${m['liq']/1000:.0f}k, h1 {m['pct_h1']:+.1f}%, h6 {m['pct_h6']:+.1f}%, buys {m['buys']}/{m['buys']+m['sells']}={100*m['buys']/max(m['buys']+m['sells'],1):.0f}%, age {m['age_h']:.1f}h, pool {pool_id}")

# Save summary
out = {
    "verdict": verdict,
    "pre_gate": pre_gate,
    "post_gate": len(survivors),
    "gate_rejections": gate_rejections,
    "src_status": src_status,
    "top15": [
        {
            "rank": i+1,
            "name": p["attributes"].get("name"),
            "network": p.get("relationships", {}).get("network", {}).get("data", {}).get("id") or (p.get("id","").split("_")[0] if "_" in p.get("id","") else None),
            "pool_id": p.get("id"),
            "tag": p["_meta"]["tag"],
            "score": round(p["_meta"]["score"], 2),
            "pct_h24": round(p["_meta"]["pct_h24"], 1),
            "pct_h1": round(p["_meta"]["pct_h1"], 1),
            "pct_h6": round(p["_meta"]["pct_h6"], 1),
            "vol_h24": int(p["_meta"]["vol_h24"]),
            "liq": int(p["_meta"]["liq"]),
            "buys": p["_meta"]["buys"],
            "sells": p["_meta"]["sells"],
            "age_h": round(p["_meta"]["age_h"], 1),
        }
        for i, p in enumerate(survivors[:15])
    ]
}
with open(os.path.join(TMPDIR, "summary.json"), "w") as f:
    json.dump(out, f, indent=2)
print("\nWrote summary.json")

File: .gt-tmp/score.js
const fs = require('fs');
const path = require('path');

const TMPDIR = '/home/runner/work/aeon/aeon/.gt-tmp';
const NOW = new Date('2026-05-29T22:14:00Z');

const FILES = {
  'gt-global.json': 'global',
  'gt-new.json': 'new',
  'gt-solana-trend.json': 'sol-trend',
  'gt-solana-vol.json': 'sol-vol',
  'gt-eth-trend.json': 'eth-trend',
  'gt-eth-vol.json': 'eth-vol',
  'gt-base-trend.json': 'base-trend',
  'gt-base-vol.json': 'base-vol',
  'gt-bsc-trend.json': 'bsc-trend',
  'gt-bsc-vol.json': 'bsc-vol',
  'gt-arbitrum-trend.json': 'arb-trend',
  'gt-arbitrum-vol.json': 'arb-vol',
};

const allPools = [];
const srcStatus = {};
for (const [fname, label] of Object.entries(FILES)) {
  const p = path.join(TMPDIR, fname);
  if (!fs.existsSync(p)) { srcStatus[label] = 'missing'; continue; }
  try {
    const data = JSON.parse(fs.readFileSync(p, 'utf8'));
    if (!data.data) { srcStatus[label] = 'fail'; continue; }
    for (const pool of data.data) {
      pool._source = label;
      allPools.push(pool);
    }
    srcStatus[label] = 'ok';
  } catch (e) {
    srcStatus[label] = 'err';
  }
}

const okSrc = Object.values(srcStatus).filter(s => s === 'ok').length;
console.log(`Loaded ${allPools.length} raw pool entries from ${okSrc}/${Object.keys(FILES).length} sources`);
console.log('Source status:', JSON.stringify(srcStatus));

const fnum = (v) => {
  if (v === null || v === undefined) return 0;
  const n = parseFloat(v);
  return isNaN(n) ? 0 : n;
};
const inum = (v) => Math.floor(fnum(v));

// Dedup by base_token
const byToken = new Map();
for (const p of allPools) {
  const bt = p.relationships?.base_token?.data?.id;
  if (!bt) continue;
  const vol = fnum(p.attributes?.volume_usd?.h24);
  const existing = byToken.get(bt);
  if (!existing || vol > fnum(existing.attributes?.volume_usd?.h24)) {
    byToken.set(bt, p);
  }
}
console.log(`After dedup: ${byToken.size} unique tokens`);
const preGate = byToken.size;

// Gate
const gateRej = { 'thin-vol': 0, negative: 0, 'thin-liq': 0, dumping: 0, honeypot: 0, 'too-new': 0, 'rug-like': 0 };
const survivors = [];
for (const p of byToken.values()) {
  const a = p.attributes || {};
  const vol = fnum(a.volume_usd?.h24);
  const pct = fnum(a.price_change_percentage?.h24);
  const liq = fnum(a.reserve_in_usd);
  const buys = inum(a.transactions?.h24?.buys);
  const sells = inum(a.transactions?.h24?.sells);
  let ageH = 9999;
  if (a.pool_created_at) {
    const dt = new Date(a.pool_created_at);
    ageH = (NOW - dt) / 3600000;
  }
  if (vol < 50000) { gateRej['thin-vol']++; continue; }
  if (pct <= 0) { gateRej.negative++; continue; }
  if (liq < 10000) { gateRej['thin-liq']++; continue; }
  if (buys > 0 && sells / Math.max(buys, 1) > 10) { gateRej.dumping++; continue; }
  if (sells > 0 && buys / Math.max(sells, 1) > 50) { gateRej.honeypot++; continue; }
  if (ageH < 1.0 && vol < 100000) { gateRej['too-new']++; continue; }
  if (pct > 10000) { gateRej['rug-like']++; continue; }

  p._meta = { vol, pct, liq, buys, sells, ageH };
  p._meta.pctH1 = fnum(a.price_change_percentage?.h1);
  p._meta.pctH6 = fnum(a.price_change_percentage?.h6);
  survivors.push(p);
}

console.log(`Post-gate: ${survivors.length} pools`);
console.log('Gate rejections:', JSON.stringify(gateRej));

const clamp = (x, lo, hi) => Math.max(lo, Math.min(hi, x));
for (const p of survivors) {
  const m = p._meta;
  const pctPts = clamp(m.pct / 500, 0, 1);
  const volPts = clamp(Math.log10(m.vol + 1) / 7, 0, 1);
  const liqPts = clamp(Math.log10(m.liq + 1) / 6, 0, 1);
  const momPts = clamp((m.pctH1 + 50) / 100, 0, 1);
  const skewPts = clamp(m.buys / Math.max(m.buys + m.sells, 1), 0, 1);
  m.score = 40 * pctPts + 25 * volPts + 15 * liqPts + 10 * momPts + 10 * skewPts;
}

for (const p of survivors) {
  const m = p._meta;
  if (m.liq >= 1_000_000 && m.vol >= 1_000_000) m.tag = 'DEEP-LIQ';
  else if (m.ageH <= 48 && m.vol >= 250_000) m.tag = 'BREAKOUT';
  else if (m.pctH1 > 2 && m.pct > 50) m.tag = 'CONTINUATION';
  else if (m.pctH1 < -5 && m.pct > 0) m.tag = 'REVERSAL';
  else m.tag = 'MICRO-SPEC';
}

survivors.sort((a, b) => b._meta.score - a._meta.score);

const top5 = survivors.slice(0, 5);
const deepCount = top5.filter(p => p._meta.tag === 'DEEP-LIQ').length;
const contCount = top5.filter(p => p._meta.tag === 'CONTINUATION').length;
let verdict;
if (survivors.length < 5) verdict = 'SLEEPY';
else if (deepCount >= 2) verdict = 'STRONG';
else if (deepCount === 1 || contCount >= 2) verdict = 'MIXED';
else verdict = 'SPECULATIVE';

console.log(`\n=== VERDICT: ${verdict} ===`);
console.log(`Top 5 distribution: DEEP-LIQ=${deepCount}, CONTINUATION=${contCount}, BREAKOUT+MICRO=${top5.filter(p => p._meta.tag === 'BREAKOUT' || p._meta.tag === 'MICRO-SPEC').length}, REVERSAL=${top5.filter(p => p._meta.tag === 'REVERSAL').length}`);

console.log('\n=== TOP 20 ===');
for (let i = 0; i < Math.min(20, survivors.length); i++) {
  const p = survivors[i];
  const m = p._meta;
  const name = p.attributes?.name || '?';
  const pid = p.id || '?';
  const network = pid.split('_')[0];
  const buyPct = Math.round(100 * m.buys / Math.max(m.buys + m.sells, 1));
  console.log(
    `${(i + 1).toString().padStart(2)}. [${m.tag.padEnd(12)}] ${name.padEnd(35)} (${network.padEnd(9)}) ` +
    `+${m.pct.toFixed(0).padStart(5)}% — score ${m.score.toFixed(1).padStart(5)}, ` +
    `vol $${(m.vol / 1e6).toFixed(1)}m, liq $${(m.liq / 1000).toFixed(0)}k, ` +
    `h1 ${m.pctH1.toFixed(1).padStart(6)}%, h6 ${m.pctH6.toFixed(1).padStart(6)}%, ` +
    `buys ${buyPct}%, age ${m.ageH.toFixed(1)}h, pool ${pid}`
  );
}

const out = {
  verdict,
  preGate,
  postGate: survivors.length,
  gateRejections: gateRej,
  srcStatus,
  top20: survivors.slice(0, 20).map((p, i) => ({
    rank: i + 1,
    name: p.attributes?.name,
    network: (p.id || '').split('_')[0],
    poolId: p.id,
    tag: p._meta.tag,
    score: Math.round(p._meta.score * 100) / 100,
    pctH24: Math.round(p._meta.pct * 10) / 10,
    pctH1: Math.round(p._meta.pctH1 * 10) / 10,
    pctH6: Math.round(p._meta.pctH6 * 10) / 10,
    volH24: Math.floor(p._meta.vol),
    liq: Math.floor(p._meta.liq),
    buys: p._meta.buys,
    sells: p._meta.sells,
    ageH: Math.round(p._meta.ageH * 10) / 10,
  })),
};
fs.writeFileSync(path.join(TMPDIR, 'summary.json'), JSON.stringify(out, null, 2));
console.log('\nWrote summary.json');
